### ISSUE TITLE
README's core_fraction coupling, f = (V_core/V_tot)**(1/3), has an infinite derivative at zero core volume

### ISSUE THREAD

**Issue by duncanwp** [reflective-org/neuralmie-jax#5]
The README's units table tells a consumer how to reach `core_fraction` from what a model actually carries:

> `core_fraction` is a **radius** ratio, so the volume fraction is its cube — `f = 0.2` is only 0.8% core by volume. Invert with `f = (V_core/V_tot)**(1/3)`.

The conversion is right, but written that way it is not safe to differentiate, and this package's stated purpose is to be used inside differentiable models.

### The failure

`d/dx x^(1/3)` is unbounded at `x = 0`, and a host evaluates this expression for every mode in every cell — including the many cells and modes carrying **no** core material at all. Zero core volume is the common case, not an edge case: whether a mode contains a core species is usually static configuration, while whether a given column contains any of it is not.

```python
import jax, jax.numpy as jnp

def f_of_vcore(v_core, v_tot=1.0):
    return jnp.cbrt(jnp.clip(v_core / v_tot, 0.0, 1.0))   # as documented

print(float(f_of_vcore(0.0)))                 # 0.0          -- value is fine
print(float(jax.grad(f_of_vcore)(0.0)))       # inf

def guarded(v_core):                          # a downstream guard does not help
    return jnp.where(v_core > 0.0, f_of_vcore(v_core), 0.0)

print(float(jax.grad(guarded)(0.0)))          # nan
```

The third line is the point. Masking the *result* cannot repair it: reverse mode multiplies the zero cotangent into the `inf` the other branch already produced, and the `NaN` then propagates through everything downstream of the optics — including derivatives with respect to a *different* species' mass in the same mode, which are perfectly well-defined. In a GCM this shows up as a whole-model NaN gradient with no finite forward symptom, which is expensive to localise.

The fix is to substitute **before** the root rather than after it:

```python
def safe(v_core, v_tot, eps=1e-30):
    ok = v_core > eps
    return jnp.where(ok, jnp.cbrt(jnp.where(ok, v_core / jnp.where(ok, v_tot, 1.0), 1.0)), 0.0)

print(float(jax.grad(lambda v: safe(v, 1.0))(0.0)))   # 0.0
```

### Suggested

Ship the conversion rather than describing it — something like

```python
core_fraction_from_volume(v_core, v_total)   # -> r_core/r_total, clipped to CORE_FRACTION_RANGE
```

with the substitution built in, an explicit and documented zero-core derivative policy, and a test at exactly zero core volume. That puts the guard in the one place that knows about `CORE_FRACTION_RANGE`'s 0.98 cap anyway, and means every consumer gets it right by default. At minimum, replace the naive expression in the README with the guarded one and say why.

Found via climate-analytics-lab/jax-gcm#793, where a downstream integration followed the documented form and produced NaN mass gradients for every core-free mode; reproduced here against this package's documented coupling alone.
